test/test_other.py

```diff
@@ -9438,6 +9438,22 @@ def test_exceptions_c_linker(self):
     stderr = self.expect_fail([EMCC, '-sSTRICT', test_file('other/test_exceptions_c_linker.c')])
     self.assertContained('error: undefined symbol: __cxa_find_matching_catch_1', stderr)
 
+  def test_exceptions_mismatch(self):
+    create_file('main.c', '''
+    #include <setjmp.h>
+    int main() {
+      jmp_buf buf;
+      longjmp(buf, 8);
+    }
+    ''')
+    self.emcc('main.c', ['-c'])
+    err = self.expect_fail([EMCC, 'main.o', '-fwasm-exceptions'])
+    self.assertContained('error: undefined reference to `emscripten_longjmp`. One or more object files was not compiled with `-fwasm-exceptions`.  Build with `-sASSERTIONS=0` to have wasm-ld report which one.', err)
+
+    err = self.expect_fail([EMCC, 'main.o', '-fwasm-exceptions', '-sASSERTIONS=0'])
+    self.assertContained('wasm-ld: error: main.o: undefined symbol: emscripten_longjmp', err)
```

Comment:
Looks like a big improvement to me.

When Emscripten invokes wasm-ld it would be nice if it could detect this error and add the extra context about this since one doesn't see the instruction to pass `-fwasm-exceptions` as a cflag here. I assume we are already redirecting stderr when we invoke `wasm-ld`?

If it's too inconvenient this by itself is quite helpful.

Reply:
Emscripten doesn't currently capture and parse the output of wasm-ld.  Its just fails if/when wasm-ld fails.

I'm not sure I want to go down the route of trying to get fancy with parsing the error output of wasm-ld.    I decided that showing the more specific error when `-O0` / `-sASSERTIONS=1` are enabled was a good compromise.